### PR TITLE
[fix](jdbc catalog) fix and add mysql and doris extremum test

### PR DESCRIPTION
**Pull request by zy-kkk**
1、add mysql jdbc catalog extremum test
2、mysql jdbc catalog time data outside the 24-hour range can be read
3、add doris jdbc catalog extremum test
4、fix doris jdbc catalog char type max size match
5、fix doris jdbc catalog nested array read

Legacy:
Doris float extreme values ​​are incorrect 3.4028235e+38, -3.4028235e+38
Correct values ​​should be 3.4028234e+38, -3.4028234e+38. The incorrect value will exceed the Java float range, so ignore this test and add it after it is fixed

This PR also followup #40122
Adds requiredColumnSize() and requiredDecimalDigits() methods to JdbcFieldSchema, improving error handling by throwing exceptions when values are absent.

### CODE REVIEW

**Comment by zy-kkk** [comment]
run buildall

File: fe/be-java-extensions/jdbc-scanner/src/main/java/org/apache/doris/jdbc/MySQLJdbcExecutor.java [hunk 1]
```diff
@@ -133,8 +134,14 @@ protected Object getColumnValue(int columnIndex, ColumnType type, String[] repla
                 case VARCHAR:
                 case ARRAY:
                     return resultSet.getObject(columnIndex + 1, String.class);
-                case STRING:
-                    return resultSet.getObject(columnIndex + 1);
+                case STRING: {
+                    int jdbcType = resultSetMetaData.getColumnType(columnIndex + 1);
+                    if (jdbcType == Types.TIME) {
```
Comment (morningman) [inline fe/be-java-extensions/jdbc-scanner/src/main/java/org/apache/doris/jdbc/MySQLJdbcExecutor.java:139]: Add some comment to explain this `if else`

File: regression-test/suites/external_table_p0/jdbc/type_test/select/test_mysql_all_types_select.groovy [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        qt_select_all_types_null """select * from all_types_nullable order by 1;"""
+
+        qt_desc_all_types_non_null """desc all_types_non_nullable;"""
+        qt_select_all_types_non_null """select * from all_types_non_nullable order by 1;"""
+
+        qt_select_varchar """select * from t_varchar order by 1;"""
+        qt_select_char """select * from t_char order by 1;"""
+
+        sql """set dry_run_query=true"""
+        qt_select_all_types_multi_block """select * from all_types_multi_block;"""
```
Comment (morningman) [inline regression-test/suites/external_table_p0/jdbc/type_test/select/test_mysql_all_types_select.groovy:49]: You can use `select count(k1), count(k2)`, and remove dry_run_query

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**